src/test/java/uk/gov/hmcts/reform/idam/client/IdamClientTest.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.hmcts.reform.idam.client;
+
+import feign.FeignException;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.contract.wiremock.AutoConfigureWireMock;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@SpringBootTest(classes = {IdamClientAutoConfiguration.class, IdamClient.class, IdamApi.class})
+@ExtendWith({
+        SpringExtension.class
+})
+@EnableAutoConfiguration
+@AutoConfigureWireMock
+class IdamClientTest {
+
+    @Autowired
+    private IdamClient idamClient;
+
+    @Test
+    @DisplayName("should return bearer token when successful")
+    void authenticateUser() {
+        String bearerToken = idamClient.authenticateUser("user@example.com", "Password12");
```

Comment:
as there is no setup stage in this test, it would be good to add a note to explain how it works. ie, how is this a valid user (and the one in test below is not).